mc-oblivious-ram/src/evictor/mod.rs

```diff
@@ -56,16 +142,125 @@ where
+        branch.ct_insert(1.into(), &stash_data[idx], &mut stash_meta[idx]);
+    }
+}
+
+pub trait BranchSelector {
+    /// Returns the leaf index of the next branch to call evict from stash
+    /// to branch on.
+    fn get_next_branch_to_evict(&mut self) -> u64;
+
+    /// Returns the number of branches to call evict from stash to branch
+    /// on.
+    fn get_number_of_additional_branches_to_evict(&self) -> usize;
+}
+
+/// Evictor trait conceptually is a mechanism for moving stash elements into
+/// the oram.
+pub trait EvictionStrategy<ValueSize, Z>
+where
+    ValueSize: ArrayLength<u8> + PartialDiv<U8> + PartialDiv<U64>,
+    Z: Unsigned + Mul<ValueSize> + Mul<MetaSize>,
+    Prod<Z, ValueSize>: ArrayLength<u8> + PartialDiv<U8>,
+    Prod<Z, MetaSize>: ArrayLength<u8> + PartialDiv<U8>,
+{
+    /// Method that takes a branch and a stash and moves elements from the
+    /// stash into the branch.
+    fn evict_from_stash_to_branch(
+        &self,
+        stash_data: &mut [A64Bytes<ValueSize>],
+        stash_meta: &mut [A8Bytes<MetaSize>],
+        branch: &mut BranchCheckout<ValueSize, Z>,
+    );
+}
+
+/// A factory which creates an Evictor
+pub trait EvictorCreator<ValueSize, Z>
+where
+    ValueSize: ArrayLength<u8> + PartialDiv<U8> + PartialDiv<U64>,
+    Z: Unsigned + Mul<ValueSize> + Mul<MetaSize>,
+    Prod<Z, ValueSize>: ArrayLength<u8> + PartialDiv<U8>,
+    Prod<Z, MetaSize>: ArrayLength<u8> + PartialDiv<U8>,
+{
+    type Output: EvictionStrategy<ValueSize, Z> + BranchSelector + Send + Sync + 'static;
+
+    fn create(&self, size: u64, height: u32) -> Self::Output;
+}
+
+/// A factory which creates an PathOramDeterministicEvictor that evicts from the
+/// stash into an additional number_of_additional_branches_to_evict branches in
+/// addition to the currently checked out branch in reverse lexicographic order
+pub struct PathOramDeterministicEvictCreator {
+    number_of_additional_branches_to_evict: usize,
+}
+impl PathOramDeterministicEvictCreator {
+    /// Create a factory for a deterministic branch selector that will evict
+    /// number_of_additional_branches_to_evict branches per access
+    pub fn new(number_of_additional_branches_to_evict: usize) -> Self {
+        Self {
+            number_of_additional_branches_to_evict,
         }
     }
 }
-impl PathOramEvict {
-    pub fn new() -> Self {
-        Self {}
+
+impl<ValueSize, Z> EvictorCreator<ValueSize, Z> for PathOramDeterministicEvictCreator
+where
+    ValueSize: ArrayLength<u8> + PartialDiv<U8> + PartialDiv<U64>,
+    Z: Unsigned + Mul<ValueSize> + Mul<MetaSize>,
+    Prod<Z, ValueSize>: ArrayLength<u8> + PartialDiv<U8>,
+    Prod<Z, MetaSize>: ArrayLength<u8> + PartialDiv<U8>,
+{
+    type Output = PathOramDeterministicEvict;
+
+    fn create(&self, size: u64, height: u32) -> Self::Output {
+        PathOramDeterministicEvict::new(self.number_of_additional_branches_to_evict, height, size)
+    }
+}
+
+#[cfg(test)]
+mod internal_tests {
```

Comment:
:upside_down_face: 
It seems that most rust code uses `mod tests` when there is only one test group inside of a file.
```suggestion
mod tests {
```

Reply:
Done.